<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>945</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="lockOrder_groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>240</width>
      <height>100</height>
     </rect>
    </property>
    <property name="title">
     <string>Lock Order</string>
    </property>
    <widget class="QRadioButton" name="radioButton_lockOrderSame">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>180</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Acquire in same order</string>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">buttonGroup_lockOrder</string>
     </attribute>
    </widget>
    <widget class="QRadioButton" name="radioButton_lockOrderInverted">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>180</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Acquire in inverted order</string>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">buttonGroup_lockOrder</string>
     </attribute>
    </widget>
   </widget>
   <widget class="QPushButton" name="lockOrder_pushButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>120</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Test Lock Order</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textBrowser_events">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>270</y>
      <width>1260</width>
      <height>631</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="textBrowser_events_label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>240</y>
      <width>56</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Events</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="sync_groupBox">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>10</y>
      <width>240</width>
      <height>100</height>
     </rect>
    </property>
    <property name="title">
     <string>Sync</string>
    </property>
    <widget class="QRadioButton" name="radioButton_syncLatch">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>180</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Sync using a Latch</string>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">buttonGroup_sync</string>
     </attribute>
    </widget>
    <widget class="QRadioButton" name="radioButton_syncBarrier">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>180</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Sync using a Barrier</string>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">buttonGroup_sync</string>
     </attribute>
    </widget>
   </widget>
   <widget class="QPushButton" name="sync_pushButton">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>120</y>
      <width>120</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Test Sync</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="promise_groupBox">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>10</y>
      <width>240</width>
      <height>100</height>
     </rect>
    </property>
    <property name="title">
     <string>Promise</string>
    </property>
    <widget class="QRadioButton" name="radioButton_matrixCalc">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>180</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Matrix Calculation</string>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">buttonGroup_promise</string>
     </attribute>
    </widget>
    <widget class="QRadioButton" name="radioButton_withException">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>180</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>With Exception</string>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">buttonGroup_promise</string>
     </attribute>
    </widget>
   </widget>
   <widget class="QPushButton" name="promise_pushButton">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>120</y>
      <width>120</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Test Promise</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>radioButton_lockOrderSame</tabstop>
  <tabstop>radioButton_lockOrderInverted</tabstop>
  <tabstop>lockOrder_pushButton</tabstop>
  <tabstop>textBrowser_events</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup_lockOrder"/>
  <buttongroup name="buttonGroup_sync"/>
  <buttongroup name="buttonGroup_promise"/>
 </buttongroups>
</ui>
